File: twitter-fetcher/src/test/java/org/jesperancinha/twitter/examples/AppleServiceTest.java
package org.jesperancinha.twitter.examples;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class AppleServiceTest {

    @Test
    void givenApples_when3Fridge6Market_thenReturn9() {
        final AppleService appleServiceMock = new AppleService(){
            @Override
            public int getAppleFromFrige() {
                return 3;
            }

            @Override
            public int getAppleFromSupermarket() {
                return 6;
            }
        };

        assertEquals(9, appleServiceMock.getTotalApples());
    }
}

File: news-cast-explorer-fetcher/src/test/kotlin/org/jesperancinha/newscast/examples/AppleServiceTest.kt
package org.jesperancinha.newscast.examples

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class AppleServiceTest {
    @Test
    fun givenApples_when3Fridge6Market_thenReturn9() {
        val appleServiceMock: AppleService = object : AppleService() {
            override val appleFromFrige: Int
                get() = 3
            override val appleFromSupermarket: Int
                get() = 6
        }
        Assertions.assertEquals(9, appleServiceMock.totalApples)
    }
}